ui-v2/tests/steps.js

```diff
@@ -0,0 +1,232 @@
+        Object.keys(data).forEach(function(key, i, arr) {
+          assert.equal(
+            body[key],
+            data[key],
+            `Expected the payload to contain ${key} to equal ${body[key]} but ${key} was ${
+              data[key]
+            }`
+          );
+        });
+      })
+      .then('a $method request is made to "$url" with the body "$body"', function(
+        method,
+        url,
+        data
+      ) {
+        const request = api.server.history[api.server.history.length - 2];
+        assert.equal(
+          request.method,
+          method,
+          `Expected the request method to be ${method} but was ${request.method}`
+        );
+        assert.equal(
+          request.url,
+          url,
+          `Expected the request url to be ${url} but was ${request.url}`
+        );
+        const body = request.requestBody;
+        assert.equal(
+          body,
+          data,
+          `Expected the request body to be ${body} but was ${request.requestBody}`
+        );
+      })
+      .then('a $method request is made to "$url"', function(method, url) {
+        const request = api.server.history[api.server.history.length - 2];
+        assert.equal(
+          request.method,
+          method,
+          `Expected the request method to be ${method} but was ${request.method}`
+        );
+        assert.equal(
+          request.url,
+          url,
+          `Expected the request url to be ${url} but was ${request.url}`
+        );
+      })
+      .then('the url should be $url', function(url) {
+        const current = currentURL();
+        assert.equal(current, url, `Expected the url to be ${url} but was ${current}`);
+      })
+      .then(['I see $num $model', 'I see $num $model model', 'I see $num $model models'], function(
+        num,
+        model
+      ) {
+        const len = currentPage[`${model}s`].filter(function(item) {
+          return item.isVisible;
+        }).length;
+
+        assert.equal(len, num, `Expected ${num} ${model}s but saw ${len}`);
+      })
+      .then(['I see $num $model model with the $property "$value"'], function(
+        num,
+        model,
+        property,
+        value
+      ) {
+        const len = currentPage[`${model}s`].filter(function(item) {
+          return item.isVisible && item[property] == value;
+        }).length;
+        assert.equal(
+          len,
+          num,
+          `Expected ${num} ${model}s with ${property} set to "${value}" but saw ${len}`
+        );
+      })
+      .then(['I see $property on the $component'], function(property, component) {
+        assert.ok(currentPage[component][property], `Expected to see ${property} on ${component}`);
+      })
+      .then(['I see $property'], function(property, component) {
+        assert.ok(currentPage[property], `Expected to see ${property}`);
+      })
+      .then('ok', function() {
+        assert.ok(true);
+      })
+  );
+}
```

Comment:
It's neat how well this all plays with ember-cli-page-object.

Reply:
👍  Yeah, in my past life the idea of page objects for acceptance were an essential building block.

Did you see it also plays well with mirage if you want it to?

Reply:
Yeah, reading over the bits for creating models and asserting against Pretender looks like it could easily be adapted to work with Mirage.